src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -144,13 +144,14 @@ static async Task<char> GetInsertionCharacterAsync(Document document, int positi
         public static Uri GetUriFromFilePath(string? filePath)
         {
             if (filePath is null)
-            {
                 throw new ArgumentNullException(nameof(filePath));
-            }
 
             return new Uri(filePath, UriKind.Absolute);
         }
 
+        public static Uri? TryGetUriFromFilePath(string? filePath)
```

Comment:
Previously if we had an invalid URI, this would throw and the exception would then be shown in the LSP logs, but now nothing about the failure would get logged.  Would it be possible to log a warning here when we are not able to parse the URI (e.g. what was the string we couldn't parse)

Reply:
sure!